385/Assets/Scripts/RopeSystem.cs

```diff
@@ -114,6 +114,15 @@ public bool IsRopeConnected()
 		return (RopeAnchorPoint != null);
     }
 
+    // gets anchor point rope is attached to
+    public Rigidbody2D getAnchor()
+    {
+        if (IsRopeConnected()) {
+            return RopeAnchorPoint;

```

Comment:
If RopeAnchorPoint is null, then it will indicate that it isn't connected.

What you are basically doing here is:
```csharp
if (RopeAnchorPoint != null) return RopeAnchorPoint;
return null;
```

which still is redundant because `RopeAnchorPoint` is publicly accessible anyways (so that it can be accessed by the unity editor. In your other scripts, I'd just reference `RopeAnchorPoint` directly.

Reply:
Now uses RopeAchorPoint